src/components/FileUpload/FileUpload.tsx

```diff
@@ -0,0 +1,485 @@
+`;
+
+const DocumentIcon = styled(Icon)`
+  svg {
+    width: ${({ theme }) => theme.click.fileUpload.sm.icon.size.width};
+    height: ${({ theme }) => theme.click.fileUpload.sm.icon.size.height};
+    color: ${({ theme }) => theme.click.fileUpload.sm.color.icon.default};
+  }
+`;
+
+const UploadIcon = styled(Icon)`
+  svg {
+    width: ${({ theme }) => theme.click.fileUpload.md.icon.size.width};
+    height: ${({ theme }) => theme.click.fileUpload.md.icon.size.height};
+    color: ${({ theme }) => theme.click.fileUpload.md.color.icon.default};
+  }
+`;
+
+const UploadText = styled.div<{ $size: "sm" | "md"; $hasFile: boolean }>`
+  text-align: ${props => (props.$hasFile || props.$size === "sm" ? "left" : "center")};
+  ${props =>
+    (props.$hasFile || props.$size === "sm") &&
+    css`
+      flex: 1;
+    `}
+
+  ${props =>
+    !props.$hasFile &&
+    props.$size === "md" &&
+    css`
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+      width: 100%;
+    `}
+`;
+
+const FileInfoHeader = styled.div`
+  display: flex;
+  align-items: center;
+  gap: ${({ theme }) => theme.click.fileUpload.sm.space.gap};
+  width: 100%;
+`;
+
+const FileInfo = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: ${({ theme }) => theme.click.fileUpload.hasFile.header.space.gap};
+  flex: 1;
+`;
+
+const FileDetails = styled.div`
+  display: flex;
+  gap: ${({ theme }) => theme.click.fileUpload.md.space.gap};
+  border: none;
+`;
+
+const FileActions = styled.div`
+  display: flex;
+  align-items: center;
+  margin-left: auto;
+  gap: 0;
+`;
+
+const ProgressContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: space-between;
+  width: 100%;
+`;
+
+const ProgressBarContainer = styled.div`
+  width: 100%;
+  flex: 1;
+`;
+
+const ProgressPercentage = styled(Text)`
+  min-width: ${({ theme }) => theme.sizes[10]};
+  text-align: right;
+  padding-right: ${({ theme }) => theme.click.fileUpload.md.space.gap};
+`;
+
+const formatFileSize = (sizeInBytes: number): string => {
+  if (sizeInBytes < 1024) {
+    return `${sizeInBytes.toFixed(1)} B`;
+  } else if (sizeInBytes < 1024 * 1024) {
+    return `${(sizeInBytes / 1024).toFixed(1)} KB`;
+  } else if (sizeInBytes < 1024 * 1024 * 1024) {
+    return `${(sizeInBytes / (1024 * 1024)).toFixed(1)} MB`;
+  } else {
+    return `${(sizeInBytes / (1024 * 1024 * 1024)).toFixed(1)} GB`;
+  }
+};
+
+const truncateFilename = (filename: string, maxLength: number = 40): string => {
+  if (filename.length <= maxLength) {
+    return filename;
+  }
+
+  const extension =
+    filename.lastIndexOf(".") !== -1 ? filename.slice(filename.lastIndexOf(".")) : "";
+
+  const nameWithoutExtension = extension
+    ? filename.slice(0, filename.lastIndexOf("."))
+    : filename;
+
+  const truncatedName =
+    nameWithoutExtension.slice(0, maxLength - extension.length - 3) + "...";
+  return truncatedName + extension;
```

Comment:
sooo this would actually produce 4 dots at the end of a truncated filename
`verylongfilena....png`
is that what we want? Looks a bit weird. I believe a common way to deal with it is to either use a different character (`verylongfilena~.png`) or to keep a few chars at the end (`verylongfil...me.png`)

Reply:
and another thing, the way it's currently implemented, it will produce weird output for hidden files like `.gitignore`, because of the leading dot
```ts
truncateFilename('.gitignore', 8) // => ....gitignore
```
they aren't the most commonly uploaded files, but it's still a use case

Reply:
do we maybe want to move these service functions to utils and cover them with unit tests?

Reply:
That would indeed be wise - I will follow up.